internal/pkg/agent/operation/operator.go

```diff
@@ -115,12 +117,45 @@ func NewOperator(
 
 	operator.initHandlerMap()
 
-	os.MkdirAll(config.DownloadConfig.TargetDirectory, 0755)
-	os.MkdirAll(config.DownloadConfig.InstallPath, 0755)
+	_ = os.MkdirAll(config.DownloadConfig.TargetDirectory, 0755)
+	_ = os.MkdirAll(config.DownloadConfig.InstallPath, 0755)
 
 	return operator, nil
 }
 
+func (o *Operator) Reload(rawConfig *config.Config) error {
+	// save some unpacking in downloaders
+	type reloadConfig struct {
+		C *artifact.Config `json:"agent.download" config:"agent.download"`
+	}
+	tmp := &reloadConfig{
+		C: artifact.DefaultConfig(),
+	}
+	if err := rawConfig.Unpack(&tmp); err != nil {
+		return errors.New(err, "failed to unpack artifact config")
+	}
+
+	r, ok := o.downloader.(artifact.ConfigReloader)
+	if !ok {
+		return nil
+	}
+
+	if err := r.Reload(tmp.C); err != nil {
+		return errors.New(err, "failed reloading downloader config")
+	}
+
+	r, ok = o.verifier.(artifact.ConfigReloader)
+	if !ok {
+		return nil
+	}
```

Comment:
good point there can be combination of non reloadable verifier or downloader, in case downloader is non reloadable this will not have desired effect. i like debug log point